cmd/maya-exporter/app/command/command.go

```diff
@@ -129,13 +135,33 @@ func (o *VolumeExporterOptions) RegisterJivaStatsExporter() error {
 	return nil
 }
 
-// RegisterCstorStatsExporter initiates the connection with the cstor and register
+// RegisterCstor initiates the connection with the cstor and register
 // the exporter with Prometheus for collecting the metrics.This doesn't returns
 // error because that case is handled in InitiateConnection().
-func (o *VolumeExporterOptions) RegisterCstorStatsExporter() {
+func (o *VolumeExporterOptions) RegisterCstor() {
 	cstor := collector.Cstor(socketPath)
 	exporter := collector.New(cstor)
 	prometheus.MustRegister(exporter)
 	glog.Info("Registered maya exporter for cstor")
 	return
 }
+
+// RegisterPool registers pool collector which collects
+// pool level metrics
+func (o *VolumeExporterOptions) RegisterPool() {
+	pool.InitVar()
+	zvol.InitVar()
+	p := pool.New()
+	z := zvol.New()
+	l := zvol.NewVolumeList()
+
+	// Blocking call with retry timeout of 2s and context timeout of 5 sec,
+	// pool container may be starting.
+	p.GetInitStatus(5 * time.Second)
```

Comment:
when this function is going on, will there be any requests to pool-exporter?

Reply:
Even if any request comes, they will get an error since server is not started yet.